app/controllers/apps_controller.rb

```diff
@@ -90,60 +90,60 @@ def regenerate_api_key
-        email_at_notices = val.gsub(/-\d+/,"").split(",").map{|v| v.strip.to_i }.reject{|v| v == 0}
-        if email_at_notices.any?
-          params[:app][:email_at_notices] = email_at_notices
-        else
-          default_array = params[:app][:email_at_notices] = Errbit::Config.email_at_notices
-          flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
-        end
+  # email_at_notices is edited as a string, and stored as an array.
+  def parse_email_at_notices_or_set_default
+    if params[:app] && (val = params[:app][:email_at_notices])
+      # Sanitize negative values, split on comma,
+      # strip, parse as integer, remove all '0's.
+      # If empty, set as default and show an error message.
+      email_at_notices = val.gsub(/-\d+/, "").split(",").map{|v| v.strip.to_i }.reject{|v| v == 0}
+      if email_at_notices.any?
+        params[:app][:email_at_notices] = email_at_notices
+      else
+        default_array = params[:app][:email_at_notices] = Errbit::Config.email_at_notices
+        flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
       end
     end
+  end
 
-    def parse_notice_at_notices_or_set_default
-      if params[:app][:notification_service_attributes] && (val = params[:app][:notification_service_attributes][:notify_at_notices])
-        # Sanitize negative values, split on comma,
-        # strip, parse as integer, remove all '0's.
-        # If empty, set as default and show an error message.
-        notify_at_notices = val.gsub(/-\d+/,"").split(",").map{|v| v.strip.to_i }
-        if notify_at_notices.any?
-          params[:app][:notification_service_attributes][:notify_at_notices] = notify_at_notices
-        else
-          default_array = params[:app][:notification_service_attributes][:notify_at_notices] = Errbit::Config.notify_at_notices
-          flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
-        end
+  def parse_notice_at_notices_or_set_default
+    if params[:app][:notification_service_attributes] && (val = params[:app][:notification_service_attributes][:notify_at_notices])
+      # Sanitize negative values, split on comma,
+      # strip, parse as integer, remove all '0's.
+      # If empty, set as default and show an error message.
+      notify_at_notices = val.gsub(/-\d+/, "").split(",").map{|v| v.strip.to_i }
+      if notify_at_notices.any?
+        params[:app][:notification_service_attributes][:notify_at_notices] = notify_at_notices
+      else
+        default_array = params[:app][:notification_service_attributes][:notify_at_notices] = Errbit::Config.notify_at_notices
+        flash[:error] = "Couldn't parse your notification frequency. Value was reset to default (#{default_array.join(', ')})."
       end
     end
+  end
 
-  private
+private
```

Comment:
I can certainly try and get rubocop to play along. That said, the keyword does become kind of hidden in this compact style, where access modifiers are otherwise used to demarcate sections of code with different audience scope: public for the world, private/protected for internal details. By putting it on a line with a method definition, to me it communicates that it affects only this one method, but I'm sure if you add another method definition below this, it would also be `private`. That probably means the line `private def app_params` should be broken up when adding to the code, creating an extra burden of consistency on contributors.

What do you think, do we look for alternatives to what is currently here?